### PR TITLE
Fix findbugs issues

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: http-client/src/main/java/com/proofpoint/http/client/Request.java [hunk 1]
```diff
@@ -39,11 +39,11 @@
     private final BodySource bodySource;
     private final boolean followRedirects;
 
-    public Request(URI uri, String method, @Nullable ListMultimap<String, String> headers, @Nullable BodySource bodySource)
+    public Request(URI uri, String method, ListMultimap<String, String> headers, @Nullable BodySource bodySource)
```
Comment (gwittel) [inline http-client/src/main/java/com/proofpoint/http/client/Request.java]: If ```headers``` is no longer nullable, do we need to enforce it with a ```checkNotNull``` or work around with a ```firstNonNull``` given it gets blindly copied?

Comment (johngmyers) [inline http-client/src/main/java/com/proofpoint/http/client/Request.java]: `ImmutableListMultimap.copyOf()` does that. It apparently never was nullable.

Comment (gwittel) [inline http-client/src/main/java/com/proofpoint/http/client/Request.java]: ```ImmutableListMultimap.copyOf()``` Doesn't seem to do the check ([1](https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/ImmutableListMultimap.java#L253)):

```java
public static <K, V> ImmutableListMultimap<K, V> copyOf(
      Multimap<? extends K, ? extends V> multimap) {
    if (multimap.isEmpty()) {
      return of();
    }
// ....
```

Comment (johngmyers) [inline http-client/src/main/java/com/proofpoint/http/client/Request.java]: `multimap.isEmpty()` will throw NPE if `multimap` is null.

Comment (gwittel) [inline http-client/src/main/java/com/proofpoint/http/client/Request.java]: Right.  If you're fine with the implicit NPE vs the other explicit checks then this is fine.

**Comment by gwittel** [comment]
LGTM.

**Pull request merged**